ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -106,6 +109,44 @@ dropListFromImportDecl iDecl = let
     f x = x
     in f <$> iDecl
 
+resolveCompletion :: IdeState -> PluginId -> CompletionItem -> LSP.LspM Config (Either ResponseError CompletionItem)
+resolveCompletion ide _ comp@CompletionItem{_detail,_documentation,_xdata}
+  | Just resolveData <- _xdata
+  , Success (uri, NameDetails mod occ) <- fromJSON resolveData
+  , Just file <- uriToNormalizedFilePath $ toNormalizedUri uri = do
```

Comment:
I think each of these cases could probably at least log if they fail, so that we know what happened. At the moment it would be totally impossible to diagnose e.g. the JSON failing to decode, we'd just get no resolution.